packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -662,6 +711,212 @@ class SyncWatcher extends BaseWatcher {
+
+      const unsetCommits = await this.getOruToAllUnsetCommitAmount()
+      availableCredit = availableCredit.sub(unsetCommits)
+    }
+
+    if (availableCredit.lt(0)) {
+      return BigNumber.from(0)
+    }
+
+    return availableCredit
+  }
+
+  async calculatePendingAmount (destinationChainId: number) {
+    const bridge = this.bridge as L2Bridge
+    const pendingAmount = await bridge.getPendingAmountForChainId(destinationChainId)
+    return pendingAmount
+  }
+
+  public async calculateUnsetCommitAmounts (destinationChainId: number) {
+    const transferRoots = await this.db.transferRoots.getUnsetCommittedTransferRoots({
+      sourceChainId: this.chainSlugToId(this.chainSlug),
+      destinationChainId
+    })
+    let totalAmount = BigNumber.from(0)
+    for (const transferRoot of transferRoots) {
+      totalAmount = totalAmount.add(transferRoot.totalAmount)
+    }
+
+    return totalAmount
+  }
+
+  private async updateAvailableCreditMap (destinationChainId: number) {
+    const availableCredit = await this.calculateAvailableCredit(destinationChainId)
+    const destinationChain = this.chainIdToSlug(destinationChainId)
+    this.availableCredit[destinationChain] = availableCredit
+
+    if (this.s3Upload) {
+      const bonder = globalConfig.bonders[this.tokenSymbol]?.[0]
+      const availableCredit = await this.calculateAvailableCredit(destinationChainId, bonder)
+      this.s3AvailableCredit[destinationChain] = availableCredit
+    }
+  }
+
+  private async updatePendingAmountsMap (destinationChainId: number) {
+    const pendingAmount = await this.calculatePendingAmount(destinationChainId)
+    const destinationChain = this.chainIdToSlug(destinationChainId)
+    this.pendingAmounts[destinationChain] = pendingAmount
+  }
+
+  private async updateUnsetCommitAmountsMap (destinationChainId: number) {
+    const totalAmounts = await this.calculateUnsetCommitAmounts(destinationChainId)
+    const destinationChain = this.chainIdToSlug(destinationChainId)
+    this.unsetCommitAmounts[destinationChain] = totalAmounts
+  }
+
+  async syncPendingAmounts () {
+    const pendingAmounts = BigNumber.from(0)
+    const chains = await this.bridge.getChainIds()
+    for (const destinationChainId of chains) {
+      const destinationChain = this.chainIdToSlug(destinationChainId)
+      if (
+        this.chainSlug === Chain.Ethereum ||
+        this.chainSlug === destinationChain
+      ) {
+        continue
+      }
+      await this.updatePendingAmountsMap(destinationChainId)
+    }
+  }
+
+  async syncUnsetCommitAmounts () {
+    const chains = await this.bridge.getChainIds()
+    for (const destinationChainId of chains) {
+      const sourceChain = this.chainSlug
+      const destinationChain = this.chainIdToSlug(destinationChainId)
+      const shouldSkip = (
+        sourceChain === Chain.Ethereum ||
+        sourceChain === destinationChain ||
+        !this.hasSiblingWatcher(destinationChainId)
```

Comment:
why does this function have the sibling watcher check but `syncPendingAmounts()` does not?

Reply:
This method has a sibling watcher call but not syncPendingAmounts